src/Modal.js

```diff
@@ -277,6 +277,11 @@ const Modal = React.createClass({
     }
   },
 
+  componentWillUnmount() {
+    this.onHide();
+    this.props.onHide();
```

Comment:
Does it make sense to call the onHide prop. If its unmounted the caller would have chosen to not display it. In our case we use the onHide callback to set some state which causes the modal not to render. So props.onHide would be called twice then